controllers/cluster_controller.go

```diff
@@ -1289,12 +1289,21 @@ func checkCoordinatorValidity(cluster *fdbtypes.FoundationDBCluster, status *fdb
 		if process.Address == "" {
 			continue
 		}
-		address, err := fdbtypes.ParseProcessAddress(process.Address)
+
+		addresses, err := fdbtypes.ParseProcessAddressesFromCmdline(process.CommandLine)
 		if err != nil {
 			return false, false, err
 		}
 
-		if address.Flags["tls"] != cluster.Spec.MainContainer.EnableTLS {
+		hasValidAddress := false
+		for _, address := range addresses {
+			if address.Flags["tls"] == cluster.Spec.MainContainer.EnableTLS {
+				hasValidAddress = true
+				break
+			}
+		}
+
+		if !hasValidAddress {
```

Comment:
We also have to change the way how `localityInfo` get's the address otherwise we get the `TLS` address again.